src/v/cluster/controller_backend.cc

```diff
@@ -1567,15 +1581,78 @@ ss::future<std::error_code> controller_backend::shutdown_on_current_shard(
     }
 }
 
+ss::future<> controller_backend::cleanup_orphan_files(
+  model::ntp ntp, model::revision_id rev) {
+    if (_self != 0) {
+        co_return;
+    }
```

Comment:
Why do we want to remove files only on a single node ?

Reply:
It should be shard_id not node_id